tools/i18n/update-for-legacy-translations

```diff
@@ -0,0 +1,242 @@
+    # "You do not have permission to use <b>@{wildcard_mention_string}</b> mentions in this channel.": "You do not have permission to use <b>@{stream_wildcard_mention}</b> mentions in this stream.",
+    "You must be an organization administrator to create a channel without subscribing.": "You must be an organization administrator to create a stream without subscribing.",
+    "You subscribed to channel <z-stream-name></z-stream-name>.": "You subscribed to stream <z-stream-name></z-stream-name>.",
+    "You unsubscribed from channel <z-stream-name></z-stream-name>.": "You unsubscribed from stream <z-stream-name></z-stream-name>.",
+    "You're not subscribed to this channel. You will not be notified if other users reply to your message.": "You're not subscribed to this stream. You will not be notified if other users reply to your message.",
+    "Your message was sent to a channel you have muted.": "Your message was sent to a stream you have muted.",
+    "back to channels": "back to streams",
+}
+
+
+def get_json_filename(locale: str) -> str:
+    return f"locale/{locale}/translations.json"
+
+
+def get_legacy_filename(locale: str) -> str:
+    return f"locale/{locale}/legacy_stream_translations.json"
+
+
+def get_locales() -> List[str]:
+    output = check_output(["git", "ls-files", "locale"], text=True)
+    tracked_files = output.split()
+    regex = re.compile(r"locale/(\w+)/LC_MESSAGES/django.po")
+    locales = []
+    for tracked_file in tracked_files:
+        matched = regex.search(tracked_file)
+        if matched and matched.group(1) != "en_GB":
+            locales.append(matched.group(1))
+
+    return locales
+
+
+def get_translations(path: str) -> Dict[str, str]:
+    with open(path) as raw_resource_file:
+        translations = json.load(raw_resource_file)
+
+    return translations
+
+
+def update_for_legacy_stream_translations(
+    current: Dict[str, str], legacy: Dict[str, str], path: str
+) -> None:
+    number_of_updates = 0
+    updated_translations: Dict[str, str] = {}
+    for line in current:
+        # If the string has a legacy string mapped and see if it's
+        # not currently translated (e.g. an empty string), then use
+        # the legacy translated string (which might be an empty string).
+        if line in LEGACY_STRINGS_MAP and current[line] == "":
+            legacy_string = LEGACY_STRINGS_MAP[line]
+            if legacy_string in legacy:
+                updated_translations[line] = legacy[legacy_string]
+                number_of_updates += 1
+        else:
+            updated_translations[line] = current[line]
+
+    # Only replace file content if we've made any updates for legacy
+    # translated strings.
+    if number_of_updates > 0:
+        with open(path, "w") as f:
+            json.dump(updated_translations, f, ensure_ascii=False, indent=2, sort_keys=True)
+            f.write("\n")
+        print(f"Updated {number_of_updates} strings in: {path}")
```

Comment:
This seemed to be an easy way to be able to note how many strings are pending translation.